src/test/java/org/prebid/server/bidder/tripleliftnative/TripleliftNativeBidderTest.java

```diff
@@ -0,0 +1,297 @@
+import com.iab.openrtb.response.SeatBid;
+import org.junit.Before;
+import org.junit.Test;
+import org.prebid.server.VertxTest;
+import org.prebid.server.bidder.model.BidderBid;
+import org.prebid.server.bidder.model.BidderError;
+import org.prebid.server.bidder.model.HttpCall;
+import org.prebid.server.bidder.model.HttpRequest;
+import org.prebid.server.bidder.model.HttpResponse;
+import org.prebid.server.bidder.model.Result;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.proto.openrtb.ext.ExtPrebid;
+import org.prebid.server.proto.openrtb.ext.request.tripleliftnative.ExtImpTripleliftNative;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.function.Function;
+
+import static java.util.Collections.emptyMap;
+import static java.util.Collections.singletonList;
+import static java.util.Collections.singletonMap;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static org.assertj.core.api.Assertions.assertThatIllegalArgumentException;
+import static org.assertj.core.api.Assertions.tuple;
+import static org.prebid.server.proto.openrtb.ext.response.BidType.xNative;
+
+public class TripleliftNativeBidderTest extends VertxTest {
+
+    private static final String ENDPOINT_URL = "https://test.endpoint.com";
+
+    private TripleliftNativeBidder tripleliftNativeBidder;
+
+    @Before
+    public void setUp() {
+        tripleliftNativeBidder = new TripleliftNativeBidder(ENDPOINT_URL, singletonMap("publisher_whitelist", "[\"foo\",\"bar\",\"baz\"]"));
+    }
+
+    @Test
+    public void creationShouldFailOnInvalidEndpointUrl() {
+        assertThatIllegalArgumentException().isThrownBy(() -> new TripleliftNativeBidder("invalid_url", emptyMap()));
+    }
+
+    @Test
+    public void creationShouldFailOnInvalidJsonPublisherWhiteList() {
+        assertThatExceptionOfType(PreBidException.class)
+                .isThrownBy(() -> new TripleliftNativeBidder(ENDPOINT_URL, singletonMap("publisher_whitelist", "bad")));
+    }
+
+    @Test
+    public void makeHttpRequestsShouldReturnErrorWhenNoNativeTypeImp() {
+        // given
+        final BidRequest bidRequest = BidRequest.builder()
+                .imp(singletonList(Imp.builder().build()))
+                .build();
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = tripleliftNativeBidder.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).hasSize(2)
+                .extracting(BidderError::getMessage)
+                .containsOnly("Unsupported publisher for triplelift_native", "no native object specified");
+        assertThat(result.getValue()).isEmpty();
+    }
+
+    @Test
+    public void makeHttpRequestsShouldReturnErrorWhenInvCodeIsNotSpecifiedImp() {
+        // given
+        final BidRequest bidRequest = BidRequest.builder()
+                .imp(singletonList(Imp.builder()
```

Comment:
Just an advise: whenever you have a bunch of various test cases for `makeHttpRequests` it's always better to introduce fabric methods like `givenBidRequest` and `givenImp` that would contain all necessary fields as well as provide functions to alter request/imp/ext as needed. That way it would be more DRY, unless it's just a couple of tests)